File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cs
public class Solution {
    IList<int> ans = new List<int>();

    public IList<int> Postorder(Node root) {
        if(root == null) return ans;
        foreach(var child in root.children)
            Postorder(child);
        ans.Add(root.val);
        return ans;
    }
}

/*
// Definition for a Node.
public class Node {
    public int val;
    public IList<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, IList<Node> _children) {
        val = _val;
        children = _children;
    }
}
*/

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.py
class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        def traverse(root: 'Node'):
            if not root: return
            for child in root.children:
                traverse(child)
            ans.append(root.val)
        
        ans = []
        traverse(root)
        return ans


"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""